### ISSUE TITLE
BaseLine Model Training

### ISSUE THREAD

**Issue by aash1999** [pranavdhawann/spring-2026-group4#48]
### **Feature: Baseline Model Training Implementation**

**Description:**
Implement training scripts for baseline models for each data type. Each script should handle model training, evaluation, metrics logging, and visualizations according to project standards.

---

**Acceptance Criteria / Requirements:**

1. **Train Script Location & Naming:**

   * Reside in `project_root/train/`
   * Naming convention: `baseline_train_<model_used>.py`

2. **Output & Storage:**

   * Results should be stored in:
     `project_root/experiments/baseline/baseline_results_<model_used>/`
   * Save model evaluation metrics as **JSON**:

     * MAE
     * MSE
     * RMSE
     * MAPE
     * SMAPE

3. **Training Logs & Visuals:**

   * Save **epoch vs train/test loss** as PNG
   * Save **checkpoints** in the same folder
   * Save **hyperparameters** used as JSON, e.g.:

     ```json
     {"n_epochs": 10, "hidden_size": 4, ...}
     ```
   * Save **model summary**
   * Generate **visualization plots** (up to 10) showing:

     * Date vs stock price
     * Input price, predicted price, actual price

4. **Model-Specific Considerations:**

   * For time series models: standardize input values (strategy can be custom)
   * Implement **early stopping** with clear criteria

---

**Notes:**

* All outputs must follow the folder structure for reproducibility.
* Plots and metrics should be clearly labeled for ease of analysis.
* Ensure scripts are modular and easy to extend to other baseline models.

**Comment by sayampalrecha** [comment]
[results to tabnet ](https://github.com/pranavdhawann/spring-2026-group4/tree/48-baseline-model-training-tabnet/experiments/baseline/baseline_results_tabnet)
